code/app/src/main/java/com/adobe/marketing/mobile/messagingsample/MainActivity.kt

```diff
@@ -65,15 +335,125 @@ class MainActivity : AppCompatActivity() {
+                    "        \"format\": \"application/json\"\n" +
+                    "      },\n" +
+                    "      \"etag\": \"1\",\n" +
+                    "      \"schema\": \"https://ns.adobe.com/experience/offer-management/content-component-json\"\n" +
+                    "    }\n" +
+                    "  ],\n" +
+                    "  \"placement\": {\n" +
+                    "    \"id\": \"com.adobe.marketing.mobile.messagingsample\",\n" +
+                    "    \"etag\": \"1\"\n" +
+                    "  }\n" +
+                    "}"
+        )
+        val convertedPayload = payloadFormatUtil.toObjectMap(payload)
+        var listPayload: ArrayList<Map<String, Any>> = ArrayList(1)
+        listPayload.add(convertedPayload as Map<String, Any>)
+        eventData.put("payload", listPayload)
+        eventData.put("requestId", "D158979E-0506-4968-8031-17A6A8A87DA8")
+
+        val edgeResponseEvent = Event.Builder(
+            "AEP Response Event Handle",
+            "com.adobe.eventType.edge", "personalization:decisions"
+        )
+            .setEventData(eventData)
+            .build()
+        MobileCore.dispatchEvent(edgeResponseEvent, null)
+    }
+}
+
+class CustomDelegate : MessageDelegate() {
+    private var currentMessage: FullscreenMessage? = null
+    var showMessages = true
+
+    override fun shouldShowMessage(fullscreenMessage: FullscreenMessage?): Boolean {
+        if (fullscreenMessage != null) {
```

Comment:
Let's do this in more Kotlin way.
`fullscreenMessage?.also {
//Other implementation
}`